taxonomy/utils.py

```diff
@@ -81,29 +83,67 @@ def get_product_skill_model_and_identifier(product_type):
     """
     Return the Skill Model along with its identifier based on product type.
     """
-    return (ProgramSkill, 'program_uuid') if product_type == ProductTypes.Program else (CourseSkills, 'course_key')
+    product_skill = (CourseSkills, 'course_key')
+    if product_type == ProductTypes.Program:
+        product_skill = (ProgramSkill, 'program_uuid')
+    elif product_type == ProductTypes.XBlock:
+        product_skill = (XBlockSkills, 'usage_key')
+    elif product_type == ProductTypes.XBlockData:
+        product_skill = (XBlockSkillData, 'xblock_id')
+    return product_skill
 
 
-def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type):
+def _create_product_with_hash(key_or_uuid, product_type, hash_content):
     """
-    Persist the skills data in the database either for Program or Course.
+    Create or update a product with hash of the text content.
+
+    Args:
+        key_or_uuid (str): product uuid
+        product_type (ProductTypes): type of product
+        hash_content (str): hash of content
+
+    Returns:
+        product object
     """
-    skill, __ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    model, identifier = get_product_skill_model_and_identifier(product_type)
+    product, _ = model.objects.update_or_create(
+        **{identifier: key_or_uuid},
+        defaults={"hash_content": hash_content, "auto_processed": True},
```

Comment:
Looks like we are trying to create a generic method here but this function is going to raise an error for other ProductTypes other than `XBlockSkills` as these fields only exist there.

Reply:
That is true, for now only `XBlockSkills` is supported. Do you think renaming this function to `_create_xblockskill_with_hash` makes more sense?

Reply:
Yes, I agree.  _create_xblockskill_with_hash makes more sense here.

Reply:
@HammadAhmadWaqas Updated. :+1: